File: frontend/src/game/objects/brick.ts
import { ANIMS } from "../sprite_animations/animations";
import { MapIndex } from "../types";
import { BaseObj } from "./base";

export class BrickObj extends BaseObj {
    isRuining = false
    constructor(
        scene: Phaser.Scene,
        index: MapIndex) {
        super(scene, index, "items", 0, "brick");
        this.sprite.anims.play(ANIMS.item.brick)
    }
    triggerRuin(onRuinCompleteCb: (mapIndex: MapIndex) => void) {
        this.isRuining = true
        this.sprite.anims.play(ANIMS.item.brickRuin)
        this.sprite.on('animationcomplete', () => {
            this.sprite.destroy();
            onRuinCompleteCb(this.getMapIndex())
        });
    }
}

File: backend2/game/objects/brick.go
package objects

import (
	"bomberman/backend2/types"
)

type BrickObj struct {
	Base      BaseObj
	IsRuining bool
}

func NewBrickObj(index *types.MapIndex) *BrickObj {
	return &BrickObj{
		Base:      NewBaseObj(index, types.ObjBrick),
		IsRuining: false,
	}
}
